src/control/index.ts

```diff
@@ -0,0 +1,174 @@
+ */
+export class GLTFFloorControl extends Control {
+    private _map: MapGL;
+    private _pluginGLTF: GltfPlugin;
+    private _root: HTMLElement;
+    private _content: HTMLElement;
+    private _contentHome: HTMLElement;
+    private _floor:
+        | undefined
+        | { currentFloorLevelKey: number | string; floorPlanId: string | number };
+
+    private _handlers: WeakMap<ChildNode, () => void>;
+
+    constructor(plugin: GltfPlugin, map: MapGL, options: ControlOptions) {
+        super(map, content, options);
+        this._pluginGLTF = plugin;
+        this._map = map;
+        this._root = this._wrap.querySelector(`.${classes.root}`) as HTMLElement;
+        this._content = this._wrap.querySelector(`.${classes.content}`) as HTMLElement;
+        this._contentHome = this._wrap.querySelector(`.${classes.contentHome}`) as HTMLElement;
+        this._handlers = new WeakMap();
+        this._root.style.display = 'none';
+
+        plugin.on('showModelFloorPlan', this._showControl);
+        plugin.on('hideModelFloorPlan', this._hideControl);
+        plugin.on('changeModelFloorPlan', this._onLevelChange);
+    }
+
+    public destroy() {
+        this._pluginGLTF.off('showModelFloorPlan', this._showControl);
+        this._pluginGLTF.off('hideModelFloorPlan', this._hideControl);
+        this._pluginGLTF.off('changeModelFloorPlan', this._onLevelChange);
+
+        this._removeButtonsEventListeners();
+
+        super.destroy();
+    }
+
+    private _removeButtonsEventListeners = () => {
+        if (!this._content) {
+            return;
+        }
+
+        this._content.childNodes.forEach((node) => {
+            if (this._handlers.has(node)) {
+                const handler = this._handlers.get(node);
+                if (handler !== undefined) {
+                    node.removeEventListener('click', handler);
+                }
+            }
+        });
+    };
+
+    private _showControl = (ev: ModelFloorPlanShowEvent) => {
+        const { currentFloorLevelKey, floorPlanId, floorLevels } = ev;
+        this._floor = { currentFloorLevelKey, floorPlanId };
+        this._root.style.display = 'block';
+        this._content.innerHTML = '';
+        this._contentHome.innerHTML = '';
+        let currentButton: HTMLElement | undefined;
+
+        const button = document.createElement('button');
+        button.className = classes.control;
+        button.innerHTML = `<div class="${classes.label}">${icon_building}</div>`;
+        button.name = 'building';
+        if (currentFloorLevelKey === 'building') {
+            button.disabled = true;
+        }
+        const handler = this._controlHandler('building');
+        button.addEventListener('click', handler);
+        this._contentHome.append(button);
+
+        floorLevels.forEach(({ floorLevelKey, floorLevelName, floorLevelIcon }) => {
+            const button = document.createElement('button');
+            let buttonContext = floorLevelName;
```

Comment:
Кажется опечатка в имени переменной:
```suggestion
            let buttonContent = floorLevelName;
```